### PR TITLE
Big fat syndication PR

### PR DESCRIPTION
**Pull request by m0g**

### CODE REVIEW

File: client/app/scripts/liveblog-edit/posts.service.js [hunk 1]
```diff
@@ -107,47 +123,64 @@ define([
...
+            console.log('retrieve post');
+            return api.posts.getById(post_id)
+                .then(_completePost)
+                .then(function(post) {
+                    return post;
+                });
         }
 
         function retrievePosts(blog_id, posts_criteria) {
+            console.log('retrieve posts');
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/posts.service.js]: remove this

File: client/app/scripts/liveblog-edit/posts.service.js [hunk 2]
```diff
@@ -107,47 +123,64 @@ define([
...
+                angular.forEach(post.items, function(val) {
+                    if (post.fullDetails) {
+                        _completeUser(val.item);
+                    }
+                });
+                _completeUser(post.mainItem.item);
+
+                if (post.syndication_in && api.hasOwnProperty('syndicationIn')) {
+                    api.syndicationIn.getById(post.syndication_in).then(function(synd) {
+                        //console.log('complete user', synd.producer_blog_title);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/posts.service.js]: remove this

File: client/app/scripts/liveblog-edit/posts.service.js [hunk 3]
```diff
@@ -107,47 +123,64 @@ define([
...
+                angular.extend(post, {
+                    // add a `multiple_items` field. Can be false or a positive integer.
+                    // FIXME: left like that to support other feature, but this need to be in camelcase
+                    multiple_items: post.groups[1].refs.length > 1 ? post.groups[1].refs.length : false,
+                    // add a `mainItem` field containing the first item
+                    mainItem: post.groups[1].refs[0],
+                    items: post.groups[1].refs
+                });
+
+                //console.log('complete post', pos.particular_type);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-edit/posts.service.js]: remove this

File: client/app/scripts/liveblog-syndication/controllers/base.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                $scope.selected[$scope.entryName] = {name: '', contacts: [{}]};
+            }
+        };
+
+        $scope.render = function(newEntry) {
+            var isEditing = false;
+
+            $scope[$scope.endPoint]._items = $scope[$scope.endPoint]._items
+                .map(function(item) {
+                    console.log('map', item._id, newEntry._id);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/controllers/base.js]: remove this

File: client/app/scripts/liveblog-syndication/directives/consumer-edit.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    apiQuery.then(function(result) {
+                        notify.pop();
+                        notify.success(gettext('consumer saved.'));
+
+                        scope.onsave({ consumer: result });
+                    })
+                    .catch(function(err) {
+                        var errorMsg = gettext('Fatal error!');
+
+                        console.log('error', err);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/directives/consumer-edit.js]: remove this

File: client/app/scripts/liveblog-syndication/directives/notifications-count.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+            template: '<span ' +
+                'class="label label-info circle small notification-counter" ' +
+                'ng-if="count > 0">{{count}}</span>' +
+                '<i class="big-icon-ingest" alt="ingest"></i>',
+            link: function(scope) {
+                var ingestPanels = ['ingest', 'incoming-syndication'];
+                scope.count = 0;
+
+                scope.$on('posts', function(e, data) {
+                    console.log('incoming', data, scope.panelState);
```
Comment (nistormihai) [inline client/app/scripts/liveblog-syndication/directives/notifications-count.js]: remove this

File: server/liveblog/posts/posts.py [hunk 7]
```diff
@@ -233,16 +252,23 @@ def on_update(self, updates, original):
...
     def on_updated(self, updates, original):
         super().on_updated(updates, original)
+        out_service = get_resource_service('syndication_out')
         # invalidate cache for updated blog
         app.blog_cache.invalidate(original.get('blog'))
         # send notifications
         if updates.get('deleted', False):
+            out_service.send_syndication_post(original, action='deleted')
             push_notification('posts', deleted=True, post_id=original.get('_id'))
         # NOTE: Seems unsused, to be removed later if no bug appears.
```
Comment (m0g) [inline server/liveblog/posts/posts.py]: @mscam this should be removed

File: server/liveblog/syndication/utils.py [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+                'refs': item_refs
+            }
+        ],
+        'highlight': False,
+        'sticky': False,
+        'syndication_in': in_syndication['_id'],
+        'particular_type': 'post',
+        'post_status': post_status,
+        'producer_post_id': producer_post_id,
+        # TODO: python-eve doesn't allow to replace _updated value, as post/put/patch method force it to now()
```
Comment (m0g) [inline server/liveblog/syndication/utils.py]: This should be removed

Comment (mscam) [inline server/liveblog/syndication/utils.py]: ok, but we should keep that in mind and open an issue on python-eve if necessary

File: server/liveblog/posts/posts.py (hunk 7 above)
Comment (mscam) [inline server/liveblog/posts/posts.py]: the line that should be removed is 261, not 260

File: client/spec/consumers_spec.js [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+var contact = {
+    firstName: 'Chuck',
+    lastName: 'Norris',
+    email: 'gmail@chucknorris.com'
+};
+
+var contact2 = {
+    firstName: 'Steven',
+    lastName: 'Seagal',
+    email: 'steven.seagal@gmail.com'
```
Comment (vladnicoara) [inline client/spec/consumers_spec.js]: Not sure that having these names as contacts is a good idea

Comment (m0g) [inline client/spec/consumers_spec.js]: Any name suggestion?

**Pull request merged**